data-prepper-plugins/s3-sink/README.md

```diff
@@ -0,0 +1,68 @@
+# S3 Sink
+
+This is the Data Prepper S3 sink plugin that sends records to an S3 bucket via S3Client.
+
+The S3 sink plugin supports OpenSearch 2.0.0 and greater.
+
+## Usages
+
+The s3 sink should be configured as part of Data Prepper pipeline yaml file.
+
+## Configuration Options
+
+```
+pipeline:
+  ...
+  sink:
+    - s3:
+        aws:
+          region: us-east-1
+          sts_role_arn: arn:aws:iam::123456789012:role/Data-Prepper
+          sts_header_overrides:
+        max_connection_retries: 5
+        max_upload_retries: 5
+        bucket: bucket_name
+        key_path_prefix: 
+        object:
+          file_pattern: events-${yyyy-MM-dd}
+        threshold:
+          event_count: 2000
+          maximum_size: 50mb
+          event_collection_duration: 20s
+        codec: 
+          json:
+        temporary_storage: local_file
+```
+
+## Configuration
+
+- `aws_region` (Optional) : The AWS region to use for credentials. Defaults to [standard SDK behavior to determine the region](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/region-selection.html).
+
+- `aws_sts_role_arn` (Optional) : The AWS STS role to assume for requests to SQS and S3. Defaults to null, which will use the [standard SDK behavior for credentials](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials.html).
+
+- `aws_sts_header_overrides`  (Optional) : An optional map of header overrides to make when assuming the IAM role for the sink plugin.
+
+- `max_connection_retries` (Optional) : An integer value indicates the maximum number of times that a single request should be retried in-order to connect s3 client.
+ 
+- `max_upload_retries` (Optional) : An integer value indicates the maximum number of times that single request should be retired in-order to ingest data to amazon s3.
```

Comment:
If there is a default value it should be specified here.

Reply:
Added the default values.